src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { SortBy } from './types/SortBy';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const [query, setQuery] = useState('');
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.All);
+
+  const selectTodo = useCallback((todo: Todo | null) => {
+    setSelectedTodo(todo);
+  }, []);
+
+  const filterTodos = () => {
+    const todosByQuery = todos.filter(({ title }) => {
+      const loweredQuery = query.toLowerCase();
+      const loweredTitle = title.toLowerCase();
+
+      return loweredTitle.includes(loweredQuery);
+    });
+
+    return todosByQuery.filter(({ completed }) => {
+      switch (sortBy) {
+        case SortBy.Completed:
+          return completed;
+
+        case SortBy.Active:
+          return !completed;
+
+        default:
+          return true;
+      }
+    });
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setIsLoaded(true);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
```

Comment:
Is there any sense to throw an error if this error is already thrown? Let's show some error message to user instead. In most cases user will not open console to check if there is an error (users usually don't know what is this at all)